monkey/infection_monkey/exploit/smbexec.py

```diff
@@ -164,3 +163,9 @@ def _exploit_host(self):
         self.add_vuln_port("%s or %s" % (SmbExploiter.KNOWN_PROTOCOLS['139/SMB'][1],
                                          SmbExploiter.KNOWN_PROTOCOLS['445/SMB'][1]))
         return True
+
+    def set_vulnerable_port(self, host: VictimHost):
+        if 'tcp-445' in self.host.services:
+            self.vulnerable_port = "445"
+        elif 'tcp-139' in self.host.services:
+            self.vulnerable_port = "139"
```

Comment:
There's a bad hidden temporal coupling here. What if none of these services are in the list?

Reply:
vulnerable port remains none, which is fine